### DISCUSSION TITLE
Using storeToRefs inside of another store is not working

### DISCUSSION THREAD

**Discussion by dbzx10299** [vuejs/pinia#2633] [Help and Questions] [answered]
I am using nuxt3 and have a store `useUserStore()` and `useWishlistStore()`

In `useUserStore` I have code that looks like this

```js
export const useUserStore = defineStore('user', () => {
  const user = ref(null)

  async function getUser() {
    user.value = await fetch(/* fetch the user */)
  }

  return { user, getUser }

})
```

Then in my wishlist store `user` is `undefined`

```js
export const useWishlistStore('wishlist', () => {
  const { user } = storeToRefs(useUserStore())

  async function getWishlistItems() {
    // user is undefined
     const { data } = await fetch(`/wishlistItems?userId=${user.value.id}`)
     // ...
  }
})
```

I can do the exact same thing like this and not destructure and then `user` does have a value.

```js
export const useWishlistStore('wishlist', () => {
  const userStore = useUserStore()

  async function getWishlistItems() {

     const { data } = await fetch(`/wishlistItems?userId=${userStore.user.id}`)
     // ...
  }
})
```

What is causing this to happen? In the header component I use `getUser` so the `user` value does update, and since closures close over a variable, and not it's value, I would expect when `getUser` fires, the `user` ref updates, and then later on when `getWishlistItems` is called it should have a value for `user` but it is always undefined.

**Comment by posva** [comment, marked as answer]
This can happen when stores are being [cross used](https://pinia.vuejs.org/cookbook/composing-stores.html) because the store object is not created for one of the stores. In practice, just use the store as a whole directly, it will work in all scenarios.

**Reply by dbzx10299** [reply]
Thanks for the help, I did read that link before but I‘m still confused why the storeToRefs does not work and using the whole store does. What is going on under the hood in general terms that makes a difference?

**Reply by posva** [reply]
When stores are cross used, whichever store gets its `use...` called first will exists as a placeholder in the other store until its own setup function returns. That's why `storeToRefs()` do not work there and should be avoided altogether with cross used stores

**Reply by blouflashdb** [reply]
Is this still the case today?

**Reply by posva** [reply]
Yes and the good practice remains the same
